process/block/baseProcess.go

```diff
@@ -1566,6 +1670,38 @@ func (bp *baseProcessor) getRootHashes(currHeader data.HeaderHandler, prevHeader
 	}
 }
 
+func (bp *baseProcessor) revertAccountsStates(header data.HeaderHandler, rootHash []byte) error {
+	err := bp.accountsDB[state.UserAccountsState].RecreateTrie(rootHash)
+	if err != nil {
+		log.Debug("recreate trie with error for header",
+			"nonce", header.GetNonce(),
+			"header root hash", header.GetRootHash(),
+			"given root hash", rootHash,
+			"error", err.Error(),
+		)
+
+		return err
+	}
+
+	validatorInfo, ok := header.(data.ValidatorStatisticsInfoHandler)
+	if !ok {
+		return process.ErrWrongTypeAssertion
+	}
+
+	err = bp.accountsDB[state.UserAccountsState].RecreateTrie(validatorInfo.GetValidatorStatsRootHash())
+	if err != nil {
+		log.Debug("revert peer state with error for header",
+			"nonce", header.GetNonce(),
```

Comment:
Maybe you can add `"header root hash", header.GetRootHash()` as well

Reply:
this was copied.